### PR TITLE
Add showAboutDialog sample

### PR DESCRIPTION
**Pull request by WalidEll**
## Description

Added a sample that shows two ways to open **AboutDialog** using showAboutDialog and AboutListTile widget
## Related Issues

[#34495]

## Tests

I added the following tests:

No tests added

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by WalidEll** [comment]
@googlebot I signed it!

**Review by HansMuller — COMMENTED** [review]
Looks pretty good; just asking for a few small changes

File: packages/flutter/lib/src/material/about.dart [hunk 1]
```diff
@@ -33,6 +33,55 @@ import 'theme.dart';
...
+/// {@tool snippet --template=stateless_widget_material}
+///
+/// This sample shows two ways to open [AboutDialog]. The first one
+/// using a [AboutListTile], while the second one is by clicking a [RaisedButton].
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///     return Scaffold(
+///       appBar: AppBar(
+///         title: Text("Show About Example"),
```
[the remark this answers is not in the document]
Comment (WalidEll) [reply packages/flutter/lib/src/material/about.dart]: if single quotes used for strings, where should we use double quotes?

[the remark this answers is not in the document]
Comment (HansMuller) [reply packages/flutter/lib/src/material/about.dart]: We rarely use them. Sometimes for a quoted substring string, like: 'Shakespeare wrote "to be or not...", etc'.

**Review by HansMuller — COMMENTED** [review]
This is looking pretty good. Here's a tweaked version of the example:

```
  @override
  Widget build(BuildContext context) {
    final TextStyle textStyle = Theme.of(context).textTheme.body1;
    final List<Widget> aboutBoxChildren = <Widget>[
      SizedBox(height: 24),
      RichText(
        text: TextSpan(
          children: <TextSpan>[
            TextSpan(
              style: textStyle,
              text: 'Flutter is Google’s UI toolkit for building beautiful, '
              'natively compiled applications for mobile, web, and desktop '
              'from a single codebase. Learn more about Flutter at '
            ),
            TextSpan(
              style: textStyle.copyWith(color: Theme.of(context).accentColor),
              text: 'https://flutter.dev'
            ),
            TextSpan(
              style: textStyle,
              text: '.'
            ),
          ],
        ),
      ),
    ];

    return Scaffold(
      appBar: AppBar(
        title: Text('Show About Example'),
      ),
      drawer: Drawer(
        child: SingleChildScrollView(
          child: SafeArea(
            child: AboutListTile(
              icon: Icon(Icons.info),
              applicationIcon: FlutterLogo(),
              applicationName: 'Show About Example',
              applicationVersion: 'August 2019',
              applicationLegalese: '© 2019 The Chromium Authors',
              aboutBoxChildren: aboutBoxChildren,
            ),
          ),
        ),
      ),
      body: Center(
        child: RaisedButton(
          child: Text('Show About Example'),
          onPressed: () {
            showAboutDialog(
              context: context,
              applicationIcon: FlutterLogo(),
              applicationName: 'Show About Example',
              applicationVersion: 'August 2019',
              applicationLegalese: '© 2019 The Chromium Authors',
              children: aboutBoxChildren,
            );
          },
        ),
      ),
    );
```

**Comment by WalidEll** [comment]
Done.

**Comment by HansMuller** [comment]
LGTM

**Pull request merged**